### PR TITLE
Add FFI for dealing with Fixed Decimal precision/truncation

### PR DESCRIPTION
**Pull request by Manishearth**
fixes https://github.com/unicode-org/icu4x/issues/1472

### CODE REVIEW

File: ffi/diplomat/Cargo.toml [hunk 1]
```diff
@@ -71,8 +71,8 @@ tinystr = { version = "0.4.10", features = ["alloc"], default-features = false }
 writeable = { path = "../../utils/writeable/" }
 
 # Run `cargo make diplomat-install` to get the right diplomat binary installed
-diplomat = { git = "https://github.com/rust-diplomat/diplomat", rev = "b448e1e272b70eae1cb2e6b7d0a2f0c68905d49a" }
-diplomat-runtime = { git = "https://github.com/rust-diplomat/diplomat", rev = "b448e1e272b70eae1cb2e6b7d0a2f0c68905d49a" }
+diplomat = { git = "https://github.com/rust-diplomat/diplomat", rev = "4d8450f575a7839e3bc2d48aedae5408f1825c30" }
```
Comment (sffc) [inline ffi/diplomat/Cargo.toml:74]: Nit: Please do the Diplomat upgrade in a separate PR.  Please do not mix diplomat upgrades with other changes because (1) it makes it harder to find regressions and (2) it makes the PR harder to review.

File: ffi/diplomat/src/fixed_decimal.rs [hunk 2]
```diff
@@ -19,14 +21,69 @@ pub mod ffi {
...
         /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html) for more information.
         pub fn create(v: i32) -> Box<ICU4XFixedDecimal> {
             Box::new(ICU4XFixedDecimal(FixedDecimal::from(v)))
         }
 
+        /// Construct an [`ICU4XFixedDecimal`] from an float, with enough digits to recover
+        /// the original floating point in IEEE 754 without needing trailing zeros
+        ///
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html#method.from_f64) for more information.
+        pub fn from_float(f: f64) -> Option<Box<ICU4XFixedDecimal>> {
```
Comment (sffc) [inline ffi/diplomat/src/fixed_decimal.rs:45]: Nit: use the word "double" for f64 and "float" for f32

**Review by sffc — CHANGES_REQUESTED** [review]
LGTM except for nits

File: ffi/diplomat/Cargo.toml (hunk 1 above)
Comment (Manishearth) [reply ffi/diplomat/Cargo.toml:74]: https://github.com/unicode-org/icu4x/pull/1478

File: ffi/diplomat/src/fixed_decimal.rs (hunk 2 above)
Comment (Manishearth) [reply ffi/diplomat/src/fixed_decimal.rs:45]: Hmm, so this gets to an interesting thing: "double" vs "float" is C/C++/Java terminology, Rust just calls them floats (and JS just calls them "numbers"). I kinda feel like we should use the Rust terminology here for consistency? I'm not really opposed to using double here, I just find it a bit weird. Then again, Rust isn't calling these APIs, C++ is.

Comment (sffc) [reply ffi/diplomat/src/fixed_decimal.rs:45]: I prefer to either use C++/Java terminology with "double", or use "f64".  But "float" is really misleading because it means a 32-bit IEEE floating point in language that are going to be calling this function.

Comment (Manishearth) [reply ffi/diplomat/src/fixed_decimal.rs:45]: I'll use f64 then
